envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt

```diff
@@ -71,8 +72,10 @@ private fun getCommunicationMode(proto: Value?): CommunicationMode {
 fun Value?.toStatusCodeFilter(
     accessLogFilterFactory: AccessLogFilterFactory
 ): AccessLogFilterSettings.StatusCodeFilterSettings? {
-    val value = if (this?.stringValue != null) this.stringValue.toUpperCase() else return null
-    return accessLogFilterFactory.parseStatusCodeFilter(value)
+    return when (this?.stringValue) {
```

Comment:
I see you replaced `if` with `when`.
The Kotlin idiomatic way to implement this:

```
fun Value?.toStatusCodeFilter(accessLogFilterFactory: AccessLogFilterFactory)
    : AccessLogFilterSettings.StatusCodeFilterSettings? = this?.stringValue?.let {
    accessLogFilterFactory.parseStatusCodeFilter(it.toUpperCase())
```

Check https://kotlinlang.org/docs/reference/null-safety.html#safe-calls

Reply:
Also, there's no test case that validates null.